src/test/java/org/springframework/session/data/mongo/integration/MongoRepositoryJdkSerializationITest.java

```diff
@@ -79,14 +93,44 @@ public void findByPrincipalNameNoSecurityPrincipalNameChangeReload() throws Exce
 		assertThat(findByPrincipalName.keySet()).containsOnly(toSave.getId());
 	}
 
+	@Test
+	public void verifyContainerClassLoaderLoadedIntoConverter() {
+
+		Field converterField = ReflectionUtils.findField(JdkMongoSessionConverter.class, "deserializer");
+		ReflectionUtils.makeAccessible(converterField);
+		DeserializingConverter deserializingConverter = (DeserializingConverter) ReflectionUtils.getField(converterField, this.converter);
+
+		Field deserializerField = ReflectionUtils.findField(DeserializingConverter.class, "deserializer");
+		ReflectionUtils.makeAccessible(deserializerField);
+		DefaultDeserializer deserializer = (DefaultDeserializer) ReflectionUtils.getField(deserializerField, deserializingConverter);
+
+		Field classLoaderField = ReflectionUtils.findField(DefaultDeserializer.class, "classLoader");
+		ReflectionUtils.makeAccessible(classLoaderField);
+		ClassLoader classLoader = (ClassLoader) ReflectionUtils.getField(classLoaderField, deserializer);
+
+		assertThat(classLoader).isEqualTo(this.classLoader);
+	}
+	
 	// tag::sample[]
 	@Configuration
 	@EnableMongoHttpSession
-	static class Config extends BaseConfig {
+	static class Config extends BaseConfig implements BeanClassLoaderAware {
+
+		private ClassLoader classLoader;
 
 		@Bean
 		public AbstractMongoSessionConverter mongoSessionConverter() {
-			return new JdkMongoSessionConverter(Duration.ofMinutes(30));
+			return new JdkMongoSessionConverter(Duration.ofMinutes(30), classLoader);
+		}
+
+		@Bean
+		public ClassLoader classLoader() {
```

Comment:
I've never seen a ClassLoader exposed as a bean before. Why is it needed?

Reply:
This was a hack to support verifying the classLoader got loaded. It's rather kludgy for something I verified via inspection in the debugger.